### ISSUE TITLE
 amplify mock api: TypeError: Cannot read property 'props' of undefined

### ISSUE THREAD

**Issue by NeillGiraldo** [aws-amplify/amplify-cli#2737]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
I have GraphQL API that was created outside of Amplify. I would like to integrate the local testing option that amplify provides into our development project. However I got this error on the browser ` TypeError: Cannot read property 'props' of undefined` when I run `amplify mock api`

**Amplify CLI Version**
3.17.0

**To Reproduce**
On an empty folder I run
- amplify init
- (configure the setting of the project) - Keep in mind here I am not having any react/angular or any other app. just amplify.
- amplify add api
   - ? Please select from one of the below mentioned services: GraphQL
   - ? Provide API name: appsynctest
   - ? Choose the default authorization type for the API IAM
   - ? Do you want to configure advanced settings for the GraphQL API No
   - ? Do you have an annotated GraphQL schema? Yes
   - ? Provide your schema file path: schema.graphql
- amplify mock api
     - ? Choose the code generation language target javascript
     - ? Enter the file name pattern of graphql queries, mutations and subscriptions sr
c/graphql/**/*.js
     - ? Do you want to generate/update all possible GraphQL operations - queries, muta
tions and subscriptions Yes
     - ? Enter maximum statement depth [increase from default if your schema is deeply
nested] 2
   ✔ Generated GraphQL operations successfully and saved at src/graphql
   AppSync Mock endpoint is running at http://xx.xx.xx.xxx:20002

However when I go to the browser, it tries to load the local testing feature but then immediately it throws the following error

`Cannot read property 'props' of undefined` on ToolbarSelect.js

**Expected behavior**
I would like to be able to access the UI for testing my Graphql API locally,

**Screenshots**
![Screenshot 2019-11-14 at 14 50 09](https://user-images.githubusercontent.com/7451047/68862910-992bf780-06ee-11ea-922d-7f32ff54245e.png)

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version: v10.15.3

**Additional context**

**Comment by ammarkarachi** [comment]
@NeillGiraldo This does look like a React error but it's hard to find out where it's originating from. Likely not a CLI error.

**Comment by NeillGiraldo** [comment]
Hi @ammarkarachi, Thanks for your answer.  Yes, it is bit a bit hard but I think the problem might be in amplify-graphiql-explorer. I have tried to do a local testing using amplify-appsync-simulator & amplify-graphiql-explorer with my own Configuration. I have found that if you use 
```
  const AppSyncConfig = {
       name: 'AppSync',
       defaultAuthenticationType: { authenticationType: 
            simulator.AmplifyAppSyncSimulatorAuthenticationType.AWS_IAM},
       additionalAuthenticationProviders: [],
  }

const configuration = {
  schema: {
    content: schema
  },
  resolvers,
  mappingTemplates,
  functions: [functionsSimulation],
  dataSources: [dataSourceDDB],
  appSync: AppSyncConfig,
};

appSyncSimulator.init(configuration);
appSyncSimulator.start();

```

you get the error but If you change the AuthenticationType to 

```
const AppSyncConfig = {
  name: 'AppSync',
  defaultAuthenticationType: { authenticationType: simulator.AmplifyAppSyncSimulatorAuthenticationType.apiKey},
  additionalAuthenticationProviders: [],
  apiKey: 'da2-fakeApiId123456',
}
```

The problem disappears.

The [authenticationTypes](https://github.com/aws-amplify/amplify-cli/blob/cac07aa2b36a65b4f86bffb82ccfacf270ee7d49/packages/amplify-graphiql-explorer/src/App.tsx#L14) used in graphiQL might not match all the authentication modes , IAM is missing.

The error is being thrown [here](https://github.com/aws-amplify/amplify-cli/blob/cac07aa2b36a65b4f86bffb82ccfacf270ee7d49/packages/amplify-graphiql-explorer/src/App.tsx#L337) at some render because the select component cannot find a selected child. Hope this helps.

**Comment by ammarkarachi** [comment]
@NeillGiraldo We don't support IAM for mock. The reason is the resolution for IAM happens in the cloud and we can't simulate that on local for mock. Let me know if you have any other questions.

**Comment by nikhilbatta** [comment]
Im having this same issue where its not rendering props when I type in amplify mock

**Comment by kaustavghosh06** [comment]
I believe this issue has been fixed with PR #3285. Please let us know if you're still seeing this issue.

**Comment by NeillGiraldo** [comment]
This PR #3476 also helps to solve the problem.
